docs/src/main/sphinx/develop/supporting-merge.rst

```diff
@@ -365,43 +365,42 @@ methods.
         ConnectorSession session,
         ConnectorTableHandle tableHandle)
 
-    This method is called during query planning to get the table layout to be
-    used for rows inserted by the ``MERGE`` operation.  For some connectors,
-    this layout will be used for rows deleted as well.
+  This method is called during query planning to get the table layout to be
+  used for rows inserted by the ``MERGE`` operation. For some connectors,
+  this layout is used for rows deleted as well.
 
 * ``getUpdateLayout()``::
 
     Optional<ConnectorTableLayout> getUpdateLayout(
         ConnectorSession session,
         ConnectorTableHandle tableHandle)
 
-    This method is called during query planning to get the table layout to
-    be used for rows deleted by the ``MERGE`` operation.  If the optional
-    return value is present, the Trino engine will use the layout for
-    updated rows.  Otherwise, it will use the result of
-    ``ConnectorMetadata.getInsertLayout`` to distribute updated rows.
+  This method is called during query planning to get the table layout to be
+  used for rows deleted by the ``MERGE`` operation. If the optional return
+  value is present, the Trino engine uses the layout for updated rows.
+  Otherwise, it uses the result of ``ConnectorMetadata.getInsertLayout`` to
+  distribute updated rows.
 
 * ``beginMerge()``::
 
     ConnectorMergeTableHandle beginMerge(
          ConnectorSession session,
-         ConnectorTableHandle tableHandle,
-         MergeDetails mergeDetails)
+         ConnectorTableHandle tableHandle)
 
   As the last step in creating the ``MERGE`` execution plan, the connector's
-  ``beginMerge()`` method is called, passing the ``session``, the
-  ``tableHandle`` and the ``MergeDetails`` object.
+  ``beginMerge()`` method is called, passing the ``session``, and the
+  ``tableHandle``.
 
   ``beginMerge()`` performs any orchestration needed in the connector to
-  start processing the ``MERGE``.  This orchestration varies from connector
-  to connector.  In the case of Hive connector operating on ACID tables,
+  start processing the ``MERGE``. This orchestration varies from connector
+  to connector. In the case of Hive connector operating on transactional tables,
   for example, ``beginMerge()`` checks that the table is transactional and
   that all updated columns are writable, and starts a Hive Metastore
```

Comment:
The "and that all updated columns are writable" should be removed, since the Hive connector doesn't do that (and currently could not, since the list of updated columns is not available in the API).